shared/js/background/redirect.es6.js

```diff
@@ -10,6 +10,23 @@ var debugRequest = false
 
 trackers.loadLists()
 
+var ports = []
+function connected(p) {
+    ports[p.sender.frameId] = p;
```

Comment:
What is this used for?

Reply:
mainly to establish an array of connections between the background page and each iframe so pipes don't get crossed.  I kinda don't like the idea of having a sparsely populated array there, so I might see if there's a better approach here.

Reply:
Where is it used though?

Reply:
that's actually a good point. I had been using it in an earlier version to send messages back to the content script in the main frame of the active tab upon receiving messages from other content scripts on the page. I ended up not needing it once I started passing messages directly from iframes to the top frame.

I'll remove.